native/core/src/common/bit.rs

```diff
@@ -1019,6 +1033,12 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_read_u64() {
+        let buffer: Vec<u8> = vec![0, 1, 2, 3, 4, 5, 6, 7];
```

Comment:
why is it u8 if we test u64, should we create a vector of i64?

Reply:
Because this is a buffer. It reads 8 bytes from here